### PR TITLE
[#1883] Add new cop Lint/NestedMethodDefinition

### PR DESCRIPTION
**Pull request by ojab**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
 * Don't count required keyword args when specifying `CountKeywordArgs: false` for `ParameterLists`. ([@sumeet][])
 * [#1879](https://github.com/bbatsov/rubocop/issues/1879): Avoid auto-correcting hash with trailing comma into invalid code in `BracesAroundHashParameters`. ([@jonas054][])
+* [#1883](https://github.com/bbatsov/rubocop/pull/1883): New cop `NestedMethodDefinition` checks for method definitions inside other methods. ([@ojab][])
```
Comment (bbatsov) [inline CHANGELOG.md]: You should also add yourself to the bottom of the file.

Comment (ojab) [inline CHANGELOG.md]: ouch, missed it. Thanks.

**Comment by bbatsov** [comment]
Just one final remark - remove the PR number from the commit message. This commit doesn't refer a bug report or a feature request, so this is redundant.

**Comment by ojab** [comment]
done. Thanks for your patience :)

**Comment by bbatsov** [comment]
No problem. Btw, you still haven't updated the commit message itself.

**Comment by ojab** [comment]
Fxd.

**Comment by bbatsov** [comment]
:+1: Thanks!

**Pull request merged**